src/java/main/org/apache/zookeeper/server/NIOServerCnxnFactory.java

```diff
@@ -165,9 +168,13 @@ public void removeCnxn(NIOServerCnxn cnxn) {
             }
 
             synchronized (ipMap) {
-                Set<NIOServerCnxn> s =
-                        ipMap.get(cnxn.getSocketAddress());
-                s.remove(cnxn);
+                InetAddress addr = cnxn.getSocketAddress();
+            	if (addr != null) {
```

Comment:
coding style issue: need one more space before the if statement so it aligns with previous statement. Other than this, patch looks good.

Reply:
tabs led the bad coding style, I have removed it.